modules/sharedaddy/sharedaddy.php

```diff
@@ -188,15 +188,26 @@ function sharing_global_resources_save() {
 }
 
 function sharing_email_dialog() {
-	echo '<div class="recaptcha" id="sharing_recaptcha"></div><input type="hidden" name="recaptcha_public_key" id="recaptcha_public_key" value="'.(defined( 'RECAPTCHA_PUBLIC_KEY' ) ? esc_attr( RECAPTCHA_PUBLIC_KEY ) : '').'" />';
+	require_once plugin_dir_path( __FILE__ ) . 'recaptcha.php';
+
+	$recaptcha = new Jetpack_ReCaptcha( RECAPTCHA_PUBLIC_KEY, RECAPTCHA_PRIVATE_KEY );
+	echo $recaptcha->get_recaptcha_html(); // xss ok
 }
 
 function sharing_email_check( $true, $post, $data ) {
-	require_once plugin_dir_path( __FILE__ ).'recaptchalib.php';
+	require_once plugin_dir_path( __FILE__ ) . 'recaptcha.php';
 
-	$recaptcha_result = recaptcha_check_answer( RECAPTCHA_PRIVATE_KEY, $_SERVER["REMOTE_ADDR"], $data["recaptcha_challenge_field"], $data["recaptcha_response_field"] );
+	$recaptcha = new Jetpack_ReCaptcha( RECAPTCHA_PUBLIC_KEY, RECAPTCHA_PRIVATE_KEY );
+	$response  = '';
+	if ( isset( $_POST['g-recaptcha-response'] ) ) {
+		$response = sanitize_text_field( $_POST['g-recaptcha-response'] );
```

Comment:
We don't store the value of `g-recaptcha-response` so I think sanitizing is not needed. There won't be `<` &mdash; it will break the reCAPTCHA markup as the value is populated in invisible textarea.

I've removed the `sanitize_text_field` in https://github.com/gedex/jetpack/commit/755d68a94ea4a466508ceef0cb8774e5d7a5f1f2.